src/v/archival/archival_policy.cc

```diff
@@ -287,6 +234,136 @@ static ss::output_stream<char> make_null_output_stream() {
     return ostr;
 }
 
+ss::future<offset_to_file_pos_result> convert_begin_offset_to_file_pos(
+  model::offset begin_inclusive,
+  const ss::lw_shared_ptr<storage::segment>& segment,
```

Comment:
remember to pass parameters to coroutines by value unless there is an exceptional case or a performance concern.